backend/src/main/java/com/bakdata/conquery/models/query/queryplan/specific/ConceptNode.java

```diff
@@ -24,25 +29,41 @@
 	private boolean tableActive = false;
 	private Map<BucketId, CBlock> preCurrentRow = null;
 	private CBlock currentRow = null;
+	private boolean excludeFromSecondaryIdQuery;
 
+	private Set<SecondaryId> usedSecondaryIds = Collections.emptySet();
 
-	public ConceptNode(ConceptElement[] concepts, long requiredBits, CQTable table, QPNode child) {
+
+	public ConceptNode(ConceptElement[] concepts, long requiredBits, CQTable table, QPNode child, boolean excludeFromSecondaryIdQuery) {
 		super(child);
 		this.concepts = concepts;
 		this.requiredBits = requiredBits;
 		this.table = table;
+		this.excludeFromSecondaryIdQuery = excludeFromSecondaryIdQuery;
 	}
 
 	@Override
 	public void init(Entity entity, QueryExecutionContext context) {
 		super.init(entity, context);
-		preCurrentRow = context.getBucketManager().getEntityCBlocksForConnector(getEntity(),table.getId());
+		preCurrentRow = context.getBucketManager().getEntityCBlocksForConnector(getEntity(), table.getId());
+
+		usedSecondaryIds = Arrays.stream(table.getResolvedConnector().getTable().getColumns())
```

Comment:
In `usedSecondaryIds` nicht die vom Datenset definierten SecondaryIds speichern sondern checken, ob die von der QueryDescription übergebene vorhanden ist und ansonsten einen Fehler werfen